semantic-scope-graph/src/Data/Module.hs

```diff
@@ -0,0 +1,41 @@
+{-# LANGUAGE DeriveTraversable #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+module Data.Module
+( Module(..)
+, ModulePath
+, ModuleInfo(..)
+, moduleInfoFromSrcLoc
+, moduleInfoFromCallStack
+) where
+
+import Data.Functor.Classes
+import Data.Maybe
+import Data.Semilattice.Lower
+import Data.Text (Text)
+import GHC.Stack
+
+data Module body = Module { moduleInfo :: ModuleInfo, moduleBody :: body }
+  deriving (Eq, Foldable, Functor, Ord, Traversable)
+
+instance Show body => Show (Module body) where
+  showsPrec d Module{..} = showsBinaryWith showsPrec showsPrec "Module" d (modulePath moduleInfo) moduleBody
+
+
+type ModulePath = FilePath
+
+data ModuleInfo = ModuleInfo { modulePath :: ModulePath, moduleLanguage :: Text, moduleOid :: Text }
```

Comment:
The `moduleLanguage` field had to change to `Text` because `Data.Language` no longer exists at this level. A couple of places in Semantic needed sprucing up after this, but it wasn’t anything too onerous.

Reply:
`Data.Language` and `Data.Blob` should eventually move to `semantic-source`, but this seems a reasonable compromise given the artificially high barrier to moving new code into `semantic-source` presented by requiring hackage releases.